controllers/servicebinding_controller.go

```diff
@@ -130,6 +130,12 @@ func (r *ServiceBindingReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 
 	if serviceBinding.Status.BindingID == "" {
+		err := r.validateSecretNameIsAvailable(ctx, serviceBinding)
+		if err != nil {
+			setBlockedCondition(err.Error(), serviceBinding)
+			return ctrl.Result{}, r.updateStatusWithRetries(ctx, serviceBinding, log)
```

Comment:
didn't we say failed condition?

Reply:
that's what i thought, but i need the blocked condition to enable the user to fix it, when I set it to failure condition the user was unable to fix it because the generation is > 0 and operation is not blocked and I didn't want to start playing with the observed generation